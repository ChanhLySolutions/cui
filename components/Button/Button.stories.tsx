import { useState } from 'react';
import { Button, ButtonProps, defaultButtonProps } from './Button';

const args: ButtonProps = {
  children: 'Button',
};

const buttonStories = {
  title: 'Components/Button',
  components: Button,
  argTypes: { onClick: { action: 'clicked' } },
  args,
};

export default buttonStories;

export function PrimaryButton(props) {
  return <Button {...props}></Button>;
}
PrimaryButton.args = {
  className: defaultButtonProps.className,
};

export function ClickButtonRed(props) {
  const [clicked, setClicked] = useState(false);

  return (
    <Button
      {...props}
      onClick={() => setClicked(true)}
      style={{ color: clicked && 'red' }}
    ></Button>
  );
}
