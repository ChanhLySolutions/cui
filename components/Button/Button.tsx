import { ButtonHTMLAttributes, MouseEventHandler } from 'react';

export interface ButtonProps extends ButtonHTMLAttributes<Element> {
  onClick?: () => void;
}

export const defaultButtonProps: ButtonProps = {
  onClick: () => {},
};

export function Button(props: ButtonProps) {
  const onClick: MouseEventHandler<HTMLButtonElement> = (event) => {
    event.preventDefault();
    props.onClick?.();
  };

  return (
    <button
      className={`box-border px-4 py-1 border ${props.className || ''}`}
      {...defaultButtonProps}
      {...props}
      onClick={onClick}
      style={props.style}
    >
      {props.children}
    </button>
  );
}
