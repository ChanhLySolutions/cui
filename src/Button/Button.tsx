import classNames from 'classnames';
import { useContext } from 'react';
import { ThemeContext } from '@/contexts/ThemeContext';

export interface ButtonProps {
  className?: string;
  onClick?: (event?: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  type?: 'primary' | 'link';
  size?: 'small' | 'medium' | 'large';
  border?: boolean;
  innerRef?: React.Ref<HTMLButtonElement>;
}

export const Button: React.FC<ButtonProps> = (props) => {
  const { theme } = useContext(ThemeContext);
  const { type, border, size = theme.size, ...rest } = props;

  const className =
    props.className ||
    classNames(
      {
        'bg-primary-500 dark:hover:bg-primary-600 hover:bg-primary-400': type === 'primary',
        'text-white': type === 'primary',
        border,
      },
      !type && 'dark:text-white',
      size,
      'rounded'
    );
  return <button className={`${className} `} {...rest}></button>;
};
