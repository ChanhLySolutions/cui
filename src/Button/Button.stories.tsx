import React from 'react';
import { Meta } from '@storybook/react/types-6-0';
import { ButtonProps, Button } from './Button';

const meta: Meta = {
  title: 'Components/Button',
  component: Button,
  parameters: {
    jest: ['Header.test.tsx'],
  },
};

export function Default(props: ButtonProps) {
  return <Button {...props}>Default</Button>;
}

export default meta;
