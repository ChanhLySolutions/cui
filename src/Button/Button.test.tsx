import renderer from 'react-test-renderer';
import { Button } from './Button';

it('renders correctly', () => {
  const button = renderer.create(<Button>Button</Button>).toJSON();
  expect(button).toMatchSnapshot();
});
