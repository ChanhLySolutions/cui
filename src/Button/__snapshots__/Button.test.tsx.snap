// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="size-md border-2 dark: text-white rounded "
>
  Button
</button>
`;
